### PR TITLE
[LC-443] Introduce travis ci 

### PR DESCRIPTION
**Pull request by metalg0su**
This PR is blocked until the following PR is resolved: 
- https://github.com/icon-project/loopchain/pull/261

# What this branch do
- Apply [travis-ci](https://travis-ci.org/icon-project/loopchain) beautiful status badges to Loopchain project
- Run tests whenever the one of branches has been pushed: PR, master, develop, release

# Test environment
- python: 36 and 37
- OS: bionic (Ubuntu 18.04)

# Steps (parallel started by matrix-multiplied number of test environments)
- Install third party libraries (rabbitMQ by docker, go-lang by pre-built file)
- Install RewardCalculator (a.k.a icon_rc)
- Install related ICON Projects
- Run `$make test`

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+- make develop
+script:
+- figlet -f smslant "loopchain test"
+- which icon_rc
+- pytest $TEST_DIR -vv
+notifications:
+  slack:
+    secure: UpmFjZxDhPhF2YKcsUyHNNg41K1ql/AYZuq6KPMWFfy2WjVOtAc9mNWOFv4AG7xW942wpKAwGA8h1T4HL26SNKIiTBTxQ6a0K8x+irgwXNt3TGQmtst6MPbrAanI6288PIA06GGxSxZnCjzDf2fB4CE4b7DxHl4GlCUJX4avm1a4tIodmUp7HHGPLCNItNs+JELHxUZ/uTbn/X1EKfHR9dBjKEkJ0pfqHtOMwfE9PgRF/xZIl7OsCaGfSqeiAvkAkd6rGTN/UYm7mlKaKMCZ6O22n0mAd8icdUQoTNDWh/aVwlJ3ZErmftMqHbgSuBeu+IYc+stluh7eKHAIFIwhvy9kSDGXDXvAyIC2fvXZUVCYIvxz50C1BkbQVGCqga4Rv2AiroSFicHccHZnNmIovwLFkprxk3w/bfdhxszLxzQ0BTUJSXte1kYJWOFKfCJqxQkeYxkVV/5DldMPQ5D47ms42IW8lQMWJQzhtAGCThIIwCG/bgPfo2Kbuowp+w+MULV+5SbN07Gi6pPnXNht2pRW/opxllOITuuvk4HbFgWy/XpShl+RU4p+jwvGrC8IEVERHjQuS+i/sENCfo/0Yv71vj4kjiVNk02tzTJtIt0yeRhgAXqsnIgN8dNiTw+Sa0kIII08oIksflG62tvaF4UjcGXj4DoNO1QBon1TYgs=
+  email:
+  - jsj8830@icon.foundation
```
Comment (metalg0su) [inline .travis.yml]: Need email address to be notified

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+   - lsof 
+   - figlet 
+   - libsecp256k1-dev
+ snaps:
+  - name: go
+    confinement: classic
+
+before_install:
+- docker pull rabbitmq:3.7
+- sudo docker run -d -p 5672:5672 -h localhost --name rabbit rabbitmq:3.7
```
Comment (yakkle) [inline .travis.yml]: how work this added?
```yml
- export PATH=/snap/bin/:$PATH
```

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+- sudo install -m 755 ./bin/icon_rc /usr/local/bin
+- sudo install -m 755 ./bin/rctool /usr/local/bin
+- cd -
+ 
+install:
+- make develop
+script:
+- figlet -f smslant "loopchain test"
+- which icon_rc
+- make test
```
[the remark this answers is not in the document]
Comment (metalg0su) [reply .travis.yml]: I already tried to separate 4 jobs at first: (2 python version) * (2 tests):
- (Python36, Python 37) * (unit-test, integration-test)

But this has PRO and CONST at the same time.
- PRO
**Easy to recognize when test failure**

- CONST
**Total required time could be longer**: Basically travis-ci provides parallel execution. But I saw that not all 4 jobs were always running. Sometimes queued and pending.

So I applied matrix only on python versions.

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+- docker
+branches:
+  only:
+  - master
+  - develop
+
+addons:
+ apt:
+  update: true
+  sources:
```
[the remark this answers is not in the document]
Comment (metalg0su) [reply .travis.yml]: Thank you

File: .travis.yml (hunk 2 above)
Comment (metalg0su) [reply .travis.yml]: I suffered finding how to use snap go instead of default go.
You finally made it!

Comment (metalg0su) [reply .travis.yml]: sadly It doesn't work. I don't know exactly what I did wrong, but I checked snap go installed and `$ go version` is 1.12.x.

But when I build icon rewardcalculator, its `Makefile` try to use default `go` whose version is 1.11.x.

T.T

Comment (yakkle) [reply .travis.yml]: can you try this?
```yml
 - PATH=/snap/bin:$PATH make && make install
```

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+  - name: go
+    confinement: classic
+ 
+before_install:
+ - git clone $ICON_RC_REPO;
+ - cd rewardcalculator;
+ - git checkout develop -f;
+ - make && sudo make install
+ - cd -;
+ - icon_rc -version;
```
Comment (yakkle) [inline .travis.yml]: remove semicolon?!

Comment (metalg0su) [reply .travis.yml]: Thank you!
I tested by multi-line command and forgot to remove them.

**Pull request closed without merging**